src/librustdoc/html/markdown.rs

```diff
@@ -104,6 +104,25 @@ thread_local!(pub static PLAYGROUND: RefCell<Option<(Option<String>, String)>> =
     RefCell::new(None)
 });
 
+macro_rules! event_loop_break {
```

Comment:
Nice. I'd call it `collect_md_events`, though, and make calling it a bit more expressive, e.g.

```rust
collect_md_events!(from parser into &mut buf (toc_builder, shorter) until End(Header(_)))
```